vm/ubpf_run.c

```diff
@@ -21,29 +21,133 @@
 //#include <stdbool.h>
 //#include <stdarg.h>
 //#include <inttypes.h>
-//#include <sys/mman.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
 #include "ubpf_int.h"
 
+static void enable_instruction_count(int fd);
+static void disable_instruction_count(int fd);
+static int setup_instruction_counter(void);
+static long long get_instruction_count(int fd);
```

Comment:
Don't use forward declarations unless there is a good reason to (like a circular dependency). Better to just put the functions ahead of their use.  But I' suspect it would be best to just have the instruction counter code in its own file.